system-test/read-rows.ts

```diff
@@ -12,69 +12,76 @@
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const emits: any[] = [{name: 'request'}];
-  if (response.row_keys) {
-    emits.push.apply(emits, [
-      {name: 'response', arg: 200},
-      {
-        name: 'data',
-        arg: {chunks: response.row_keys.map(rowResponse)},
-      },
-    ]);
-  }
-  if (response.end_with_error) {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const error: any = new Error();
-    error.code = response.end_with_error;
-    emits.push({name: 'error', arg: error});
-  } else {
-    emits.push({name: 'end'});
-  }
-  let index = 0;
-  setImmediate(next);
-
-  function next() {
-    if (index < emits.length) {
-      const emit = emits[index];
-      index++;
-      emitter.emit(emit.name, emit.arg);
-      setImmediate(next);
-    }
-  }
-}
-
-function rowResponse(rowKey: {}) {
+function rowResponseFromServer(rowKey: string) {
   return {
-    rowKey: Mutation.convertToBytes(rowKey),
+    rowKey: Buffer.from(rowKey).toString('base64'),
     familyName: {value: 'family'},
-    qualifier: {value: 'qualifier'},
-    valueSize: 0,
-    timestampMicros: 0,
-    labels: [],
+    qualifier: {value: Buffer.from('qualifier').toString('base64')},
     commitRow: true,
-    value: 'value',
+    value: Buffer.from(rowKey).toString('base64'),
   };
 }
 
+function getRequestOptions(request: any): google.bigtable.v2.IRowSet {
+  const requestOptions = {} as google.bigtable.v2.IRowSet;
+  if (request.rows && request.rows.rowRanges) {
+    requestOptions.rowRanges = request.rows.rowRanges.map(
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (range: any) => {
+        const convertedRowRange = {} as {[index: string]: string};
+        {
+          // startKey and endKey get filled in during the grpc request.
+          // They should be removed as the test data does not look
+          // for these properties in the request.
+          if (range.startKey) {
+            delete range.startKey;
+          }
+          if (range.endKey) {
+            delete range.endKey;
+          }
+        }
+        Object.keys(range).forEach(
+          key => (convertedRowRange[key] = range[key].asciiSlice())
+        );
+        return convertedRowRange;
+      }
+    );
+  }
+  if (request.rows && request.rows.rowKeys) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    requestOptions.rowKeys = request.rows.rowKeys.map((rowKeys: any) =>
+      rowKeys.asciiSlice()
+    );
+  }
+  // The grpc protocol sets rowsLimit to '0' if rowsLimit is not provided in the
+  // grpc request.
+  //
+  // Do not append rowsLimit to collection of request options if received grpc
+  // rows limit is '0' so that test data in read-rows-retry-test.json remains
+  // shorter.
+  if (request.rowsLimit && request.rowsLimit !== '0') {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (requestOptions as any).rowsLimit = parseInt(request.rowsLimit);
```

Comment:
once again asking about the `any` 🙂

Reply:
This one was a good one to fix. Helped catch some incorrect types downstream.